src/kernel/boot.c

```diff
@@ -839,24 +834,63 @@ BOOT_CODE static bool_t check_available_memory(word_t n_available,
         /* Available regions must be sane */
         if (r->start > r->end) {
             printf("ERROR: memory region %"SEL4_PRIu_word" has start > end\n", i);
-            return false;
+            return 0;
         }
 
         /* Available regions can't be empty. */
         if (r->start == r->end) {
             printf("ERROR: memory region %"SEL4_PRIu_word" empty\n", i);
-            return false;
+            return 0;
         }
 
         /* Regions must be ordered and must not overlap. Regions are [start..end),
            so the == case is fine. Directly adjacent regions are allowed. */
         if ((i > 0) && (r->start < available[i - 1].end)) {
             printf("ERROR: memory region %d in wrong order\n", (int)i);
-            return false;
+            return 0;
+        }
+
+        /* Sanitize the available memory region, only such memory can be used
+         * that is visible in the kernel window. On 32-bit architectures this
+         * is not be uncommon, because the 32-bit virtual address space is too
+         * small to support a kernel windows that is big enough.
+         */
+        if ((r->start > PADDR_TOP) || (r->end <= PADDR_BASE)) {
+            printf("    can't use region, outside of kernel window\n");
```

Comment:
If I understand the boot code correctly, removing things from `freemem` will mean that they are used for device untyped. So the memory is still used, just not available for other kernel objects. 

```suggestion
            printf("    Region outside of kernel window, will use for device untypeds instead.\n");
```